fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/LazyOpener.java

```diff
@@ -99,12 +102,15 @@ public T getUnchecked() {
         try {
             return getInternal();
         } catch (ExecutionException e) {
-            if (e.getCause() instanceof UncheckedIOException) {
-                final UncheckedIOException cause = ((UncheckedIOException)e.getCause());
+            Throwable cause = e.getCause();
+            if (cause instanceof RuntimeException) {
+                // Runtime exception - just rethrow - for example, RecordCoreException, UncheckedIOException
                 cause.addSuppressed(e);
-                throw cause;
+                throw (RuntimeException)cause;
+            } else {
+                // Otherwise, wrap with generic RecordCoreException
+                throw new RecordCoreException(cause);
```

Comment:
Should this use `toRecordCoreException` if it is an `IOException`

Reply:
Done.